File: backend/spring-boot/src/main/java/org/bootstrapbugz/api/auth/jwt/redis/repository/RefreshTokenStoreRepository.java
package org.bootstrapbugz.api.auth.jwt.redis.repository;

import java.util.List;
import java.util.Optional;
import org.bootstrapbugz.api.auth.jwt.redis.model.RefreshTokenStore;
import org.springframework.data.repository.CrudRepository;

public interface RefreshTokenStoreRepository extends CrudRepository<RefreshTokenStore, String> {
  Optional<RefreshTokenStore> findByUserIdAndIpAddress(Long userId, String ipAddress);

  List<RefreshTokenStore> findAllByUserId(Long userId);
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/jwt/redis/repository/RefreshTokenStoreRepository.kt
package org.bootstrapbugz.api.auth.jwt.redis.repository

import java.util.Optional
import org.bootstrapbugz.api.auth.jwt.redis.model.RefreshTokenStore
import org.springframework.data.repository.CrudRepository

interface RefreshTokenStoreRepository : CrudRepository<RefreshTokenStore, String> {
  fun findByUserIdAndIpAddress(userId: Long?, ipAddress: String): Optional<RefreshTokenStore>

  fun findAllByUserId(userId: Long?): List<RefreshTokenStore>
}
